src/jit/codegencommon.cpp

```diff
@@ -8517,13 +8519,23 @@ void                CodeGen::genFnProlog()
     // since they are trashed by the jithelper call to setup the PINVOKE frame
     if (compiler->info.compCallUnmanaged)
     {
-        excludeMask |= (RBM_PINVOKE_FRAME | RBM_PINVOKE_TCB | RBM_PINVOKE_SCRATCH);
+        excludeMask |= RBM_PINVOKE_FRAME;
+        if (compiler->opts.eeFlags & CORJIT_FLG_PINVOKE_USE_HELPERS)
+        {
+            noway_assert(compiler->info.compLvFrameListRoot == BAD_VAR_NUM);
+        }
+        else
+        {
+            noway_assert(compiler->info.compLvFrameListRoot < compiler->lvaCount);
```

Comment:
A noway assert will be present not just in debug even in retail code.  Further noway assert when fired in retail will trigger JIT compilation in minopts.  If the same noway assert is hit even in min opts then it will lead to throwing an Invalid program exception.